### ISSUE TITLE
Documentation is useless

### ISSUE THREAD

**Issue by nicbou** [nvie/rq#903]
The first page of the docs says twice that rq was designed with a lower barrier to entry. My first impression of rq was different. I spent my first few hours digging into the source code to understand basic features that were simply undocumented:

* How to specify a different redis URL. The `-u` flag is not documented.
* How to specify the Python path. Part of the command line interface is not documented.
* How to clear a queue. There is zero mention of `Queue.empty()` anywhere.
* How to create several workers. I presume you call `rq worker` several times?
* Default values (TTL, timeout) are mentioned in passing on random pages, not where the Job attributes are listed.
* Job attributes are not listed on the Jobs page, where one would expect to find them.
* Most of the Worker API is undocumented.

We are currently evaluating different task queues for a project, and the quality of the documentation is the biggest point against rq. It's not an appropriate level of documentation for a framework that claims a low barrier of entry.

**Comment by selwin** [comment]
Thanks for the feedback.

Yes, I do understand that the documentation is sorely lacking. This is partly due to the fact that whenever I find the time to maintain this project, I spend most of it in code rather than documentation because it's not something I'm good at.

The list you gave is an excellent starting point - I'll try to cover some of that sometime in the next few days.

Hope is always welcome :)

**Comment by rminderhoud** [comment]
I agree with @nicbou, the documentation is a good start but trying to do anything even a bit more complex required me to jump into the source. Thankfully the source is high quality and easy to follow.

@selwin - I think an "easy" low-hanging fruit would be to incorporate a doc generator that will at least list public class/methods/properties for the main types (queue, job, etc.)

**Comment by kevinkle** [comment]
Thought I'd add, we've found it useful to search old issues for features we ended up using but weren't documented. For reference (no promises, holidays and all), would PR'ing against https://github.com/nvie/rq/tree/gh-pages be the way to go for contribs to this? I should note, we may not be implementing things as they should be.

**Comment by selwin** [comment]
@kevinkle yes, definitely. Please open a PR for gh-pages branch and I'll be more than happy to review them.

**Comment by nicbou** [comment]
@kevinkle, this is also how I dealt with most issues, although I often ended up reading the source directly.

@selwin, I totally understand your perspective. I'm not currently using RQ (we picked something else at work, and I'm done with my personal project). However, if I pick it up again, I'll create documentation as I go. As mentioned before, a doc generator would already help a lot.
